include/hdlConvertor/hdlObjects/hdlOperatorType.h

```diff
@@ -4,16 +4,12 @@ namespace hdlConvertor {
 namespace hdlObjects {
 
 enum HdlOperatorType {
-	RANGE, // range used in VHDL type specifications
-	RANGE_REVERSE, // reverse range used in VHDL type specifications
 	ACROSS,
 	THROUGH,
 	REFERENCE,
 	TOLERANCE,
 	TYPE_OF, // SV type operator
 	INDEX, // array index
-	DOWNTO, // downto for the slice specification
```

Comment:
It seems to me that because `HdlRange, HdlConstraint, HdlSubtype...` does not inherit from `iHldExpr` things are getting complicated. Is this because you are trying to ensure all the places where this object are used are handled correctly or it is because iHdlExpr current impl. does not allow that?

Reply:
Yes, there is some complexity in disambiguating these concepts.  If everything is one (base) type, then things get easier.  For example, Python's `list` and `dict` types store `PyObject` instances.  And since everything inherits from `PyObject`, those data types can store anything.

But a range, constraint, or subtype are not an "expression" in the usual, mathematical sense.  And other than a few corner cases, splitting these out wasn't too hard.  I left most of the `src/vhdl/exprParser.cpp` methods returning `std::unique_ptr<iHdlExpr>`.  The ones that were returning subtypes or ranges I modified appropriately, and the boundaries between the concepts came out fairly clean.

My main question right now is whether I should collapse all of the constraints into one class, all of the type declarations into one class, and all of the ranges into one class.  That would save lots of work in the Python "toX" regime.  It would be similar to `HdlValue` and `iHdlExpr`. 
Thoughts?

Reply:
> all of the constraints into one class

How many types of constraints do we have? 
```
      range_constraint
      array_constraint
      record_constraint
```
?

Reply:
That's right.  A simple range constraint, an array constraint (index and element), and record constraint (some/all fields constrained).  In another implementation I also had a "no constraint" for when I parsed `OPEN`.

Reply:
In that case let's have them as a separate classes with a common interface (oop). Maybe also the we want to use some different *_constraint from *_constraint hierarchy as main representatives. Do as you find suitable. We do not want to tons of code because of maintenance cost but we do not want to lose the information.

Reply:
OK, thanks.  I'm going to compromise between my original approach and "everything is an `iHdlExpr`" approach.  My original approach had distinct classes for those different kinds of constraints, different kinds of type declarations, and different types of ranges.  I think I'm going to keep the concepts of "subtype", "type declaration", "constraint", and "range", and make the distinctions between the different types using a 'type' field like you did or just by which member variables are populated.  Thoughts?